examples/alphaTestExamples/pruneNamespace.md

```diff
@@ -0,0 +1,184 @@
+```
+
+Apply the "app" to the cluster. All the config maps should be created, and
+no resources should be pruned.
+<!-- @runApply @testE2EAgainstLatestRelease -->
+```
+kapply apply $BASE --reconcile-timeout=1m > $OUTPUT/status
+expectedOutputLine "namespace/test-namespace created"
+expectedOutputLine "configmap/cm-a created"
+expectedOutputLine "configmap/cm-b created"
+expectedOutputLine "configmap/cm-c created"
+expectedOutputLine "4 resource(s) applied. 4 created, 0 unchanged, 0 configured"
+expectedOutputLine "0 resource(s) pruned, 0 skipped"
+
+# There should be only one inventory object
+kubectl get cm --selector='cli-utils.sigs.k8s.io/inventory-id' --no-headers | wc -l > $OUTPUT/status
+expectedOutputLine "1"
+# Capture the inventory object name for later testing
+invName=$(kubectl get cm --selector='cli-utils.sigs.k8s.io/inventory-id' --no-headers | awk '{print $1}')
+# There should be four config maps: one inventory in default, two in test-namespace, one in default namespace
+kubectl get cm --selector='cli-utils.sigs.k8s.io/inventory-id' --no-headers | wc -l > $OUTPUT/status
+expectedOutputLine "1"
+kubectl get cm -n test-namespace --selector='!cli-utils.sigs.k8s.io/inventory-id' --no-headers | wc -l > $OUTPUT/status
+expectedOutputLine "2"
+kubectl get cm --selector='!cli-utils.sigs.k8s.io/inventory-id' --no-headers | wc -l > $OUTPUT/status
+expectedOutputLine "1"
+# ConfigMap cm-a had been created in the cluster
+kubectl get configmap/cm-a -n test-namespace --no-headers | wc -l > $OUTPUT/status
+expectedOutputLine "1"
+# ConfigMap cm-b had been created in the cluster
+kubectl get configmap/cm-b --no-headers | wc -l > $OUTPUT/status
+expectedOutputLine "1"
+# ConfigMap cm-c had been created in the cluster
+kubectl get configmap/cm-c -n test-namespace --no-headers | wc -l > $OUTPUT/status
+expectedOutputLine "1"
+```
+
+## Update the "app" to remove a two of the config maps, and the
+namespace.
+
+Remove test-namespace
+Remove cm-a
+Remove cm-b
+
+<!-- @createAnotherConfigMap @testE2EAgainstLatestRelease -->
+```
+
+rm -f $BASE/namespace.yaml
+rm -f $BASE/config-map-a.yaml
+rm -f $BASE/config-map-b.yaml
+
+```
+
+## Apply the updated "app"
+
+cm-a should be pruned (since it has been deleted locally).
+cm-b should be pruned (since it has been deleted locally).
+test-namespace should **not** be pruned.
+
+<!-- @applySecondTime @testE2EAgainstLatestRelease -->
+```
+kapply apply $BASE --reconcile-timeout=1m > $OUTPUT/status
+expectedOutputLine "configmap/cm-a pruned"
+expectedOutputLine "configmap/cm-b pruned"
+expectedOutputLine "configmap/cm-c unchanged"
+expectedOutputLine "2 resource(s) pruned, 1 skipped"
+
+# The test-namespace should not be pruned.
+kubectl get ns test-namespace --no-headers | wc -l > $OUTPUT/status
+expectedOutputLine "1"
+# There should be only one inventory object
+kubectl get cm --selector='cli-utils.sigs.k8s.io/inventory-id' --no-headers | wc -l > $OUTPUT/status
```

Comment:
I think we also need to verify the namespace "test-namespace" is still in the inventory list.

Reply:
No. Currently any resource that is skipped in the prune is removed from the inventory. So the `test-namespace` will not be in the inventory list. 

It sounds like we want to change this (including for resources that are prune-skipped because of the `on-remove: keep` annotation). But that will happen in a future PR.

Reply:
I changed that in #283, the skipped namespace and the skipped resource due to lifecycle annotation are both added to the inventory list.

Reply:
Ok--nice. Let me try to update the e2e test.